elide-core/src/main/java/com/yahoo/elide/graphql/Entity.java

```diff
@@ -0,0 +1,152 @@
+/* Copyright 2017, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.graphql;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.RequestScope;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+public class Entity {
+    private Optional<Entity> parentResource;
+    private Optional<Map<String, Object>> data;
+    private Class<?> entityClass;
+
+    /**
+     * Class constructor
+     * @param parentResource parent entity
+     * @param data entity data
+     * @param entityClass binding entity class
+     */
+    protected Entity(Optional<Entity> parentResource, Optional<Map<String, Object>> data, Class<?> entityClass) {
+        this.parentResource = parentResource;
+        this.data = data;
+        this.entityClass = entityClass;
+    }
+
+    /**
+     * Getter for binding entity class
+     */
+    public Class<?> getEntityClass() {
+        return this.entityClass;
+    }
+
+    /**
+     * Getter for entity data
+     */
+    public Map<String, Object> getData() {
+        return this.data.orElse(null);
+    }
+
+    /**
+     * Getter for parent entity
+     */
+    public Optional<Entity> getParentResource() {
+        return parentResource;
+    }
+
+    /**
+     * Get the id of the entity
+     * @param requestScope Request scope
+     * @return the optional id
+     */
+    public Optional<String> getId(RequestScope requestScope) {
```

Comment:
Since you aren't using the entire RequestScope, you should change this to just dictionary.